rllib/core/models/tests/test_recurrent_encoders.py

```diff
@@ -0,0 +1,89 @@
+import itertools
+import unittest
+
+from ray.rllib.core.models.base import ENCODER_OUT, STATE_OUT
+from ray.rllib.core.models.configs import LSTMEncoderConfig
+from ray.rllib.utils.test_utils import framework_iterator, ModelChecker
+
+
+class TestRecurrentEncoders(unittest.TestCase):
+    def test_lstm_encoders(self):
+        """Tests building LSTM encoders properly and checks for correct architecture."""
+
+        # Loop through different combinations of hyperparameters.
+        inputs_dimss = [[1], [100]]
+        output_dimss = [[1], [100]]
+        num_lstm_layerss = [1, 3]
+        hidden_dims = [16, 128]
+        output_activations = [None, "linear", "relu"]
+        use_biases = [False, True]
+
+        for permutation in itertools.product(
+            inputs_dimss,
+            num_lstm_layerss,
+            hidden_dims,
+            output_activations,
+            output_dimss,
+            use_biases,
+        ):
+            (
+                inputs_dims,
+                num_lstm_layers,
+                hidden_dim,
+                output_activation,
+                output_dims,
+                use_bias,
+            ) = permutation
+
+            print(
+                f"Testing ...\n"
+                f"input_dims: {inputs_dims}\n"
+                f"num_lstm_layers: {num_lstm_layers}\n"
+                f"hidden_dim: {hidden_dim}\n"
+                f"output_activation: {output_activation}\n"
+                f"output_dims: {output_dims}\n"
+                f"use_bias: {use_bias}\n"
+            )
+
+            config = LSTMEncoderConfig(
+                input_dims=inputs_dims,
+                num_lstm_layers=num_lstm_layers,
+                hidden_dim=hidden_dim,
```

Comment:
Not that I think that it's needed, but is there a reason why we only offer a single hidden dim as opposed to making this more flexible with removing `num_lstm_layers` and making `hidden_dim` `hidden_dims`?

Reply:
Is it "just" because it makes things ugly and complicated and is not really needed?

Reply:
Good question. Yeah, I did this for simplification reasons. We might want to generalize this in the future, but I'll leave it as-is for now. I'll add a TODO.

Reply:
Actually, no, torch only allows a single hidden dim value for all sub-layers within a torch.nn.LSTM. I'll just leave it at that.